apps/openmw/mwsound/sound.hpp

```diff
@@ -11,85 +11,67 @@ namespace MWSound
     inline int operator&(int a, PlayMode b) { return a & static_cast<int>(b); }
     inline int operator&(PlayMode a, PlayMode b) { return static_cast<int>(a) & static_cast<int>(b); }
 
+    struct SoundParams
+    {
+        osg::Vec3f mPos;
+        float mVolumeFactor = 1;
```

Comment:
if I remembered correctly, some developers complained about such approach due to implicit type conversions.

Reply:
But it's absolutely valid thing. There is no precision loss. What is the problem with this? Why `1.f` is better?

Reply:
Check the discussion at #2882. It is not my complaint - your approach is pretty fine for me.

Reply:
So it's only an opinion that I simply don't agree with.

Reply:
It wasn't a complaint, it was a suggestion by @Capostrophic though no one else commented further on it. The rest of the conversation instead started discussing about the proper way to skin a cat as there are quite a few ways to do so. 

float mVolumeFactor = 1.f;
or
float mVolumeFactor = 1;

Is fine, there is no reason to be nit-picky here as it is evaluated the same anyway.

Reply:
The discussion was over `{...}` uniform initialization syntax, to ensure no implicit conversion or narrowing occurs, rather than `= ...` initialization syntax. Whether it's `= 1` or `= -1.f` on its own doesn't matter much, because the compiler will likely optimize it either way (though uniform initialization syntax does help guarantee the right thing happens). Regardless, what's acceptable depends on what the project's coding style guidelines say.

Reply:
More to the patch itself, I don't think there's a need for a third volume field. Since `mBaseVolume` is always the result of `volumeFromType(type)` and the sound params hold the sound type, it can be easily retrieved without being explicitly stored with the sound. The additional `mVolumeFactor` can just take over `mBaseVolume` instead. So the fields would be:
```
float mVolume; // Volume as set in the sound definition
float mBaseVolume; // Caller's volume parameter
...
int mFlags; // Contains the sound type, for volumeFromType
...
float getRealVolume() const noexcept
{
    return mParams.mVolume * mParams.mBaseVolume * volumeFromType(getType(mParams.mFlags));
}
```

Reply:
Right, so this is a non-issue. Moving on.